cassandra_test.go

```diff
@@ -602,6 +602,35 @@ func TestBatch(t *testing.T) {
 	}
 }
 
+func TestBatchMemAlloc(t *testing.T) {
+	session := createSession(t)
+	defer session.Close()
+
+	batch := session.NewBatch(LoggedBatch).SizeHint(10)
+	if len(batch.Entries) != 0 {
+		t.Fatalf("batch entries size want %d, got %d", 0, len(batch.Entries))
+	}
+
+	for i := 0; i < 100; i++ {
+		batch.Query(`INSERT INTO batch_table (id) VALUES (?)`, i)
+	}
+	if len(batch.Entries) != 100 {
+		t.Fatalf("batch entries size want %d, got %d", 100, len(batch.Entries))
+	}
+
+	batch.SizeHint(10)
+	if len(batch.Entries) != 100 {
+		t.Fatalf("batch entries size want %d, got %d", 100, len(batch.Entries))
+	}
+
+	for i := 0; i < 100; i++ {
+		batch.Query(`INSERT INTO batch_table (id) VALUES (?)`, i)
+	}
+	if len(batch.Entries) != 200 {
+		t.Fatalf("batch entries size want %d, got %d", 100, len(batch.Entries))
```

Comment:
`want` should be 200 here, not 100.

Maybe the check could be refactored to a function so that the value is not repeated twice?

```go
checkLength := func(n int) {
    if len(batch.Entries) != n {
        t.Fatalf("batch entries length want %d, got %d", n, len(batch.Entries))
    }
}
```